### PR TITLE
Enable usage of ElasticGit's shiny new RemoteWorkspace

### PR DESCRIPTION
**Pull request by smn**

### CODE REVIEW

**Comment by smn** [comment]
Not at all ready for review.

**Comment by Rizziepit** [comment]
@smn @miltontony can you give me a small review or feedback? Particularly this stuff:
1. The import command can [import a repo remotely](https://github.com/universalcore/springboard/pull/25/files#diff-e2b1bd75365fcde553b76d97f40a3d4eR110) if you specify `--host`. All other commands are for local use only.
2. The search class `SM` handles remote repos using `RepoHelper` (that's a new elastic-git thing). I've added [`CachingRepoHelper`](https://github.com/universalcore/springboard/pull/25/files#diff-462c182f4f5b38aeab7fa74941115537R191) in springboard to cache active branch names for the duration of a request. Otherwise springboard makes 20+ repo GET requests to unicore.distribute for the home page.
3. How should [`api_notify`](https://github.com/universalcore/springboard/pull/25/files#diff-526c7a67294145e099a5c30d5b85920aR94) work for remote repos?

This is almost done. Just checking if I need to change direction before writing tests.

File: springboard/views/core.py [hunk 1]
```diff
@@ -93,9 +93,13 @@ def flat_page(self):
...
     def api_notify(self):
-        for working_dir, index_prefix in zip(self.all_repo_paths,
+        for working_dir, index_prefix in zip(self.all_repo_urls,
                                              self.all_index_prefixes):
-            fastforward.delay(os.path.abspath(working_dir), index_prefix)
+            if not any([
+                    working_dir.startswith('http://'),
+                    working_dir.startswith('https://')]):
+                fastforward.delay(os.path.abspath(working_dir), index_prefix)
+            # TODO: what is the best way to update a remote repo?
```
Comment (miltontony) [inline springboard/views/core.py]: It would be a POST to the repo in UCD `$ curl -XPOST http://localhost:6543/repos/unicore-sample-content.json`

**Comment by miltontony** [comment]
:+1: with just a note on the fastforward that it should do a POST to UCD

**Comment by Rizziepit** [comment]
@miltontony all done. final review?

**Comment by miltontony** [comment]
:+1:

**Pull request merged**